nomic/utils.py

```diff
@@ -233,3 +236,12 @@ def get_object_size_in_bytes(obj):
         marked.update(new_refr.keys())
 
     return sz
+
+# Helpful function for downloading feather files
+def download_feather(url: str, path: str):
+    path.parent.mkdir(parents=True, exist_ok=True)
+    with requests.get(url, stream=True) as response:
+        response.raise_for_status()
+        with open(path, 'wb') as f:
+            for chunk in response.iter_content(chunk_size=8192):
+                f.write(chunk)
```

Comment:
As a rule we *don't* want to use this function, so I think shoudl remove it. We store feather files on our web site without column level compression, but with gzip compression applied to the whole file, because the javascript arrow package does not support column level  compression. It's better to read the full arrow file into memory and then write it to disk for two reasons:
1. We can apply good zstd compression client side, using less disk space of our users;
2. It ensures that the files are valid arrow on disk, rather than let them potentially get written with a 'feather' suffix but actually be the text of a 403 error or something.

It's better to use the model in the quadtrees, because this will use zstd compression:

```python
                data = requests.get(url)
                readable = io.BytesIO(data.content)
                readable.seek(0)
                tb = feather.read_table(readable, memory_map=True)
                path.parent.mkdir(parents=True, exist_ok=True)
                feather.write_feather(tb, path)
```

BUT the one place I deliberately used the old `download_file` method was for getting sidecars, because it's possible that someone will have specifically a full text sidecar for say Obelics where even a single quadtree tile is too large to load into memory. For tags this isn't the case--we know they're small enough to load.

Reply:
A second issue here, because that one is getting long--these functions also need to include bearer tokens, don't they? Just a plain requests get won't be authenticated. I'm a little concerned in fact that the tests are *passing*--does that mean we're overly permissive in letting non-authenticated users download tags.

Reply:
Ah agreed that we need to add bearer tokens. These methods currently dont have tests as I need to create a tagged map on the test account (see todo) so no need to be concerned about passing tests.

And gotcha that makes sense on reading and writing feather.

Reply:
Adding on, this method didn't exist separately in utils until this PR. I noticed that I was using it in a few places so I pulled it out. I don't think we should remove it but keep a general method that uses the feather read and write